internal/search/query/transformer.go

```diff
@@ -563,3 +563,67 @@ func FuzzifyRegexPatterns(nodes []Node) []Node {
 		return Parameter{Field: field, Value: value, Negated: negated, Annotation: annotation}
 	})
 }
+
+// mapRevFilters removes rev: filters from []Node and attaches their value as @rev to the repo: filters.
+// All rev: filters are assumed to be on the top-level, i.e. mapRevFilter will not traverse the tree.
+// To be compatible with the output of the parser with and without DNF we handle both, []Node with
+// an explicit top-level And operator and []Node with an implicit And.
+func mapRevFilters(nodes []Node) ([]Node, error) {
```

Comment:
```suggestion
func concatRevFilters(nodes []Node) ([]Node, error) {
```

How about `concat`? It's just that `map` doesn't add much, except the notion that something is changed.